src/lib/button-toggle/button-toggle.ts

```diff
@@ -145,43 +191,25 @@ export class MatButtonToggleGroup extends _MatButtonToggleGroupMixinBase
     super();
   }
 
-  private _updateButtonToggleNames(): void {
-    if (this._buttonToggles) {
-      this._buttonToggles.forEach((toggle) => {
-        toggle.name = this._name;
-      });
-    }
+  ngOnInit() {
+    this._selectionModel = new SelectionModel<MatButtonToggle>(this.multiple, undefined, false);
   }
 
-  // TODO: Refactor into shared code with radio.
-  private _updateSelectedButtonToggleFromValue(): void {
-    let isAlreadySelected = this._selected != null && this._selected.value == this._value;
-
-    if (this._buttonToggles != null && !isAlreadySelected) {
-      let matchingButtonToggle = this._buttonToggles.filter(
-          buttonToggle => buttonToggle.value == this._value)[0];
-
-      if (matchingButtonToggle) {
-        this.selected = matchingButtonToggle;
-      } else if (this.value == null) {
-        this.selected = null;
-        this._buttonToggles.forEach(buttonToggle => {
-          buttonToggle.checked = false;
-        });
-      }
+  ngAfterContentInit() {
+    // If there was an attempt to assign a value before init, use it to set the
+    // initial selection, otherwise check the `checked` state of the toggles.
+    if (this._tempValue) {
```

Comment:
We need `this._tempValue !== null` here to accept falsy values.

Reply:
Fixed.